test/hotspot/jtreg/serviceability/jvmti/SuspendWithObjectMonitorEnter/SuspendWithObjectMonitorEnter.java

```diff
@@ -0,0 +1,406 @@
+// <launch returns>   wait for notify   contender running
+// launch resumer     :                 block on threadLock
+// <launch returns>   :                 :                    resumer running
+// suspend contender  :                 <suspended>          wait for notify
+// <ready to test>    :                 :                    :
+// :                  :                 :                    :
+// notify blocker     wait finishes     :                    :
+// notify resumer     exit threadLock   :                    wait finishes
+// join blocker       :                 :                    enter threadLock
+// <join returns>     blocker exits     <resumed>            resume contender
+// join resumer                         :                    exit threadLock
+// <join returns>                       enter threadLock     resumer exits
+// join contender                       exit threadLock
+// <join returns>                       contender exits
+//
+
+public class SuspendWithObjectMonitorEnter {
+    private static final String AGENT_LIB = "SuspendWithObjectMonitorEnter";
+    private static final int exit_delta   = 95;
+
+    private static final int DEF_TIME_MAX = 60;    // default max # secs to test
+    private static final int JOIN_MAX     = 30;    // max # secs to wait for join
+
+    public static final int THR_MAIN      = 0;     // ID for main thread
+    public static final int THR_BLOCKER   = 1;     // ID for blocker thread
+    public static final int THR_CONTENDER = 2;     // ID for contender thread
+    public static final int THR_RESUMER   = 3;     // ID for resumer thread
+
+    public static final int TS_INIT              = 1;  // initial testState
+    public static final int TS_BLOCKER_RUNNING   = 2;  // blocker is running
+    public static final int TS_CONTENDER_RUNNING = 3;  // contender is running
+    public static final int TS_RESUMER_RUNNING   = 4;  // resumer is running
+    public static final int TS_CALL_SUSPEND      = 5;  // call suspend on contender
+    public static final int TS_DONE_BLOCKING     = 6;  // done blocking threadLock
+    public static final int TS_READY_TO_RESUME   = 7;  // ready to resume contender
+    public static final int TS_CALL_RESUME       = 8;  // call resume on contender
+    public static final int TS_CONTENDER_DONE    = 9;  // contender has run; done
+
+    public static Object barrierLaunch = new Object();   // controls thread launch
+    public static Object barrierBlocker = new Object();  // controls blocker
+    public static Object barrierResumer = new Object();  // controls resumer
+    public static Object threadLock = new Object();      // testing object
+
+    public volatile static int testState;
+    public static long count = 0;
+
+    private static void log(String msg) { System.out.println(msg); }
+
+    native static int GetResult();
+    native static void SetPrintDebug();
+    native static void SuspendThread(int id, SuspendWithObjectMonitorEnterWorker thr);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Reply:
Nice catch. In the original 20 year old test those methods were:
```
    native static int diobj4413752GetMonitorInfo(int id, Object mon);
    native static int diobj4413752GetResult();
    native static int diobj4413752SuspendThread(int id, diobjWorker thr);
    native static int diobj4413752Wait4ContendedEnter(int id, diobjWorker thr);
```
so when I got rid of the `diobj4413752` prefix I forgot to fix the new first letter.
I'll take care of those method renames and I'll double check for others.